### ISSUE TITLE
Code-breaking semicolon inserted before arrow function with --no-semi

### ISSUE THREAD

**Issue by rotu** [prettier/prettier#13414]
This is very similar to #12964 and mentioned in https://github.com/prettier/prettier/issues/12964#issuecomment-1154829566.
As per https://github.com/prettier/prettier-vscode/issues/2507#issuecomment-1236157983, this was apparently *not* fixed by #13173.

When formatting a range in an arrow function body, if `--no-semi` is enabled, a semicolon is prepended to the arrow function.

**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEAzKmC8mAFEQJQEB8mwAOnpgE4YCuDUA3JgPRd0C+pdiAA0ICAAcYAS2gBnZKACGDBhADuABWUJ5KRajWKAnvNEAjBorABrDAGVFAWzgAZKVDjJs+2XHOWbe3ErdwBzZBgGZj8QOEczOAATRKSXRShQ5kVQuAAxCAZHRRhpDOQQRWYYCBEQAAsYR1QAdTqpeFlgsDg7HXapADd2o3KwWVMQd18GGA1LUKKvHxiAK1kADzsw9ABFZgh4JdRfUWCGafKzRQTUWvEGdxhmqUSYOuQADgAGU9VfZss4nK9zg0wGnlEAEd9vA5hJdBVZABaDxJJK1JjQqRMObZRZIbzHGK+RxSI4nECybZwPYHTwE5aiGDXZ6vd5IABMTMsUlQYQAwhBHPjYrIAKy1Zi+AAq110hIpA2iAEkoClYHYwA9JABBNV2GBGdDkmKWDI9ZkzZAARg+ojNOQAomrkABma18PhAA)

```sh
--parser babel
--no-semi
--range-start 18
--range-end 31
```

**Input:**

```jsx
let fn = (() => {
  return; //
});
```

**Output:**

```jsx
let fn = (;() => {
  return //
});
```

**Expected behavior:**

```jsx
let fn = (() => {
  return //
});
```

**Comment by techan-af** [comment]
Hey @rotu 

I was able to replicate this issue in the prettier playground. If you change the name of your function from "fn" to anything else, that is not a reserved keyword, it works. 

#### Input

```diff
- let fn = (() => {
+ let newFunction = (() => {
   return; //
});
```
#### Output
```js
let newFunction = (() => {
  return //
});
```

I think this is possible as "fn" maybe some kind of reserved keyword, that's why it's creating this error. I am still working on the issue and will get back to you.

**Comment by thorn0** [comment]
It's the same issue as https://github.com/prettier/prettier/issues/12964
It's been fixed, but not released yet.

**Comment by rotu** [comment]
@thorn0 d'oh! You are correct. I wrongly assumed it was released and that Playground was at HEAD.

**Comment by thorn0** [comment]
It was merged to `next`, so it'll be part of v3.0.0. No ETA, sorry.

**Comment by sfinnie5** [comment]
Any updates?
